audb/core/load.py

```diff
@@ -1037,6 +1037,129 @@ def load(
     return db
 
 
+def load_attachments(
+        name: str,
+        attachments: typing.Union[str, typing.Sequence[str]],
+        *,
+        version: str = None,
+        cache_root: str = None,
+        num_workers: typing.Optional[int] = 1,
+        timeout: float = -1,
+        verbose: bool = True,
+) -> audformat.Database:
+    r"""Load attachment(s) of database.
+
+    Args:
+        name: name of database
+        attachments: attachment ID
+            or sequence of attachment IDs
+            to load
+        version: version of database
+        cache_root: cache folder where databases are stored.
+            If not set :meth:`audb.default_cache_root` is used
+        num_workers: number of parallel jobs or 1 for sequential
+            processing. If ``None`` will be set to the number of
+            processors on the machine multiplied by 5
+        timeout: maximum wait time if another thread or process is already
+            accessing the database. If timeout is reached, ``None`` is
+            returned. If timeout < 0 the method will block until the
+            database can be accessed
+        verbose: show debug messages
+
+    Returns:
+        list of attachment file paths
+
+    Raises:
+        ValueError: if an attachment ID is requested
+            that is not part of the database
+
+    Examples:
+        >>> load_attachments(
+        ...     'emodb',
+        ...     [],
+        ...     version='1.3.0',
+        ...     verbose=False,
+        ... )
+        []
+
+    """
+    attachments = audeer.to_list(attachments)
+    if version is None:
+        version = latest_version(name)
+
+    db_root = database_cache_root(name, version, cache_root)
+
+    if verbose:  # pragma: no cover
+        print(f'Get:   {name} v{version}')
+        print(f'Cache: {db_root}')
+
+    deps = dependencies(
+        name,
+        version=version,
+        cache_root=cache_root,
+        verbose=verbose,
+    )
+    # We use single archive per attachment ID,
+    # so we can infer available attachment IDs
+    # from archive names
+    attachment_ids = list(
+        deps._df[
+            deps._df['type'] == define.DependType.ATTACHMENT
+        ].archive
+    )
+    attachment_files = []
+    for attachment in attachments:
+        if attachment not in attachment_ids:
+            msg = error_message_missing_object(
+                'attachment',
+                [attachment],
+                name,
+                version,
+            )
+            raise ValueError(msg)
+        attachment_files += list(
+            deps._df[
+                deps._df['archive'] == attachment
+            ].index
+        )
+
+    try:
+        with FolderLock(db_root):
```

Comment:
Here you are not passing the timeout parameters, but surprisingly this still does not solve the issue.

```suggestion
        with FolderLock(db_root, timeout=timeout):
```